aws_advanced_python_wrapper/host_monitoring_v2_plugin.py

```diff
@@ -0,0 +1,506 @@
+                        # Kill Connection
+                        monitor_context.set_node_unhealthy()
+                        connection_to_abort = monitor_context.get_connection()
+                        monitor_context.set_inactive()
+                        if connection_to_abort is not None:
+                            self.abort_connection(connection_to_abort)
+                            if self._aborted_connection_counter is not None:
+                                self._aborted_connection_counter.inc()
+                    elif monitor_context.is_active():
+                        temp_active_contexts.append(monitor_context)
+
+                for active_context in temp_active_contexts:
+                    self._active_contexts.put(weakref.ref(active_context))
+
+                delay_ns = self._failure_detection_interval_ns - (status_check_end_time_ns - status_check_start_time_ns)
+                if delay_ns < self._THREAD_SLEEP_NANO:
+                    delay_ns = self._THREAD_SLEEP_NANO
+
+                sleep(delay_ns // 1_000_000_000)
+
+        except InterruptedError:
+            pass
+        except Exception:
+            logger.debug("HostMonitorV2.ExceptionDuringMonitoringStop", self._host_info.host)
+        finally:
+            self.stop()
+            if self._monitoring_connection is not None:
+                try:
+                    self._monitoring_connection.close()
+                except AwsWrapperError:
+                    pass
+
+    def check_connection_status(self) -> bool:
+        connect_telemetry_context = self._telemetry_factory.open_telemetry_context("connection status check",
+                                                                                   TelemetryTraceLevel.FORCE_TOP_LEVEL)
+
+        if connect_telemetry_context is not None:
+            connect_telemetry_context.set_attribute("url", self._host_info.url)
+
+        try:
+            if self._monitoring_connection is None or self._driver_dialect.is_closed(self._monitoring_connection):
+                monitoring_properties = copy.deepcopy(self._props)
+                for property_key in monitoring_properties.keys():
+                    if property_key.startswith(self._MONITORING_PROPERTY_PREFIX):
+                        monitoring_properties[property_key[len(self._MONITORING_PROPERTY_PREFIX):]] = \
+                            monitoring_properties[property_key]
+                        monitoring_properties.pop(property_key, None)
+
+                logger.debug("HostMonitorV2.OpeningMonitoringConnection", self._host_info.url)
+                self._monitoring_connection = self._plugin_service.force_connect(self._host_info, monitoring_properties)
+                logger.debug("HostMonitorV2.OpenedMonitoringConnection", self._host_info.url)
+                return True
+
+            valid_timeout = ((self._failure_detection_interval_ns - self._THREAD_SLEEP_NANO) // 2) * 10**9
```

Comment:
I think this should be true division too?
```suggestion
            valid_timeout = ((self._failure_detection_interval_ns - self._THREAD_SLEEP_NANO) / 2) / 10**9
```